src/pubsub/ua_pubsub_manager.c

```diff
@@ -157,6 +157,19 @@ UA_Server_addPublishedDataSet(UA_Server *server,
         return result;
     }
 
+    if(UA_String_isEmpty(&publishedDataSetConfig->name))
+    {
+        // DataSet has to have a valid name
+        return result;
+    }
+
+    if(UA_PublishedDataSet_findPDSbyName(server, publishedDataSetConfig->name))
+    {
+        // DataSet name has to be unique in the publisher
```

Comment:
Please add a logging message here.
Otherwise users will take a long time to debug.